ci-operator/step-registry/assisted/baremetal/operator/catalog-publish/assisted-baremetal-operator-catalog-publish-commands.sh

```diff
@@ -0,0 +1,87 @@
+echo "   umoci installed"
+
+echo "## Install python dependencies"
+python3 -m pip install --user --upgrade pip
+python3 -m pip install --user -r deploy/olm-catalog/requirements.txt
+echo "   python dependencies installed"
+
+BUNDLE_DESTINATION="${REGISTRY_HOST}/${REGISTRY_ORG}/${REGISTRY_BUNDLE_REPOSITORY_NAME}:${REGISTRY_BUNDLE_REPOSITORY_TAG}"
+echo "## Generate and publish bundle to ${BUNDLE_DESTINATION}"
+
+SRC_DIR="${PWD}"
+
+tmp=$(mktemp -d)
+cd "${tmp}"
+
+/tmp/umoci init --layout bundle
+/tmp/umoci new --image bundle:latest
+/tmp/umoci config --config.label 'operators.operatorframework.io.bundle.mediatype.v1=registry+v1' --image bundle:latest
+/tmp/umoci config --config.label 'operators.operatorframework.io.bundle.mediatype.v1=registry+v1' --image bundle:latest
+/tmp/umoci config --config.label 'operators.operatorframework.io.bundle.manifests.v1=manifests/' --image bundle:latest
+/tmp/umoci config --config.label 'operators.operatorframework.io.bundle.metadata.v1=metadata/' --image bundle:latest
+/tmp/umoci config --config.label 'operators.operatorframework.io.bundle.package.v1=assisted-service-operator' --image bundle:latest
+/tmp/umoci config --config.label 'operators.operatorframework.io.bundle.channels.v1=alpha,ocm-2.5' --image bundle:latest
+/tmp/umoci config --config.label 'operators.operatorframework.io.bundle.channel.default.v1=alpha' --image bundle:latest
+/tmp/umoci config --config.label 'operators.operatorframework.io.metrics.builder=operator-sdk-v1.10.0+git' --image bundle:latest
+/tmp/umoci config --config.label 'operators.operatorframework.io.metrics.mediatype.v1=metrics+v1' --image bundle:latest
+/tmp/umoci config --config.label 'operators.operatorframework.io.metrics.project_layout=go.kubebuilder.io/v3' --image bundle:latest
+/tmp/umoci unpack --rootless --image bundle:latest bundle
+
+python3 "${SRC_DIR}/deploy/olm-catalog/pin-latest.py"
+cp -r "${SRC_DIR}/deploy/olm-catalog/manifests" bundle/rootfs/
+cp -r "${SRC_DIR}/deploy/olm-catalog/metadata" bundle/rootfs/
+
+/tmp/umoci repack --image bundle:latest bundle
+skopeo copy oci:bundle:latest "docker://${BUNDLE_DESTINATION}"
+
+# Get bundle digest
+BUNDLE_DIGEST=$(skopeo inspect oci:bundle:latest | jq -r '.Digest')
+BUNDLE_DESTINATION_DIGEST="${REGISTRY_HOST}/${REGISTRY_ORG}/${REGISTRY_BUNDLE_REPOSITORY_NAME}@${BUNDLE_DIGEST}"
+
+CATALOG_DESTINATION="${REGISTRY_HOST}/${REGISTRY_ORG}/${REGISTRY_CATALOG_REPOSITORY_NAME}:${REGISTRY_CATALOG_REPOSITORY_TAG}"
+echo "## Generate and publish catalog to ${CATALOG_DESTINATION}"
+
+tmp=$(mktemp -d)
+cd "${tmp}"
+
+/tmp/opm index add \
+    --bundles "${BUNDLE_DESTINATION_DIGEST}" \
+    --out-dockerfile index.Dockerfile \
+    --generate
+
+skopeo copy "docker://${OPM_BASE_IMAGE}" oci:opm:latest
+/tmp/umoci unpack --rootless --image opm:latest bundle
+mv database bundle/rootfs/
+/tmp/umoci repack --image opm:index bundle
```

Comment:
optional: you can set ``export PATH="${PATH}":/tmp/bin`` and have umochi/opm in there (to improve appearance)

Reply:
The other way around `export PATH="/tmp/bin:${PATH}"` if we want to ensure we're using the one we downloaded for ourselves

Reply:
Done